web3/providers/websocket.py

```diff
@@ -32,14 +32,15 @@ def get_default_endpoint():
 
 class PersistentWebSocket:
 
-    def __init__(self, endpoint_uri, loop):
+    def __init__(self, endpoint_uri, loop, **kwargs):
         self.ws = None
         self.endpoint_uri = endpoint_uri
         self.loop = loop
+        self.kwargs = kwargs
 
     async def __aenter__(self):
         if self.ws is None:
-            self.ws = await websockets.connect(uri=self.endpoint_uri, loop=self.loop)
+            self.ws = await websockets.connect(uri=self.endpoint_uri, loop=self.loop, **self.kwargs)
```

Comment:
Maybe guard against people passing in `loop` or `uri` as one of the `connection_params` here.